api/end_device.proto

```diff
@@ -941,3 +941,28 @@ message BatchDeleteEndDevicesRequest {
     }
   ];
 }
+
+message BatchGetEndDevicesRequest {
+  ttn.lorawan.v3.ApplicationIdentifiers application_ids = 1 [(validate.rules).message.required = true];
+  repeated string device_ids = 2 [
+    (validate.rules).repeated = {
+      min_items: 1,
+      max_items: 20,
+      items: { string: { pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$", max_len: 36 } }
+    }
+  ];
+  // The names of the end device fields that should be returned.
+  // This mask is applied on all the end devices in the result.
+  // See the API reference for which fields can be returned by the different services.
+  google.protobuf.FieldMask field_mask = 3;
+  // Order the results by this field path (must be present in the field mask).
+  // Default ordering is by ID. Prepend with a minus (-) to reverse the order.
+  string order = 4 [
+    (validate.rules).string = { in: ["", "device_id", "-device_id", "join_eui", "-join_eui", "dev_eui", "-dev_eui",
+      "name", "-name", "description", "-description", "created_at", "-created_at", "last_seen_at", "-last_seen_at"] }
+  ];
+  // Limit the number of results per page.
```

Comment:
It's a bit weird that we have pagination in a request where the identifiers have been fully provided to be honest (and the limit is also small).

Reply:
Ok understood. 

Do you want me to remove the ordering as well?

> Should we just keep these fields as reserved (id + name) ?

Do you mean keep the api fields reserved but not use it?

Reply:
The backend ordering matters a lot for pagination, since you cannot sort without having all of the pages available as a client. I think the ordering is not needed either.

For future proofing in general we reserve the field numbers and the field names, such that in the future we can come back and add the functionality if needed. You can see an example of this [here](https://github.com/TheThingsNetwork/lorawan-stack/blob/7d93026eecbb8f20133794d5a375a93a8681a677/api/identityserver.proto#L124).